### PR TITLE
Misc Axis Refactor

### PR DESCRIPTION
**Pull request by kanitw**
- [x] simplify labelAngle / labelBaseline

### CODE REVIEW

File: src/compile/axis/config.ts [hunk 1]
```diff
@@ -1,14 +1,71 @@
...
 
+function getAxisConfigFromConfigTypes(configTypes: string[], config: Config) {
+  // TODO: add special casing to add conditional value based on orient signal
+  return Object.assign.apply(null, [{}, ...configTypes.map(configType => config[configType])]);
+}
+
+export type AxisConfigs = ReturnType<typeof getAxisConfigs>;
+
+export function getAxisConfigs(channel: PositionScaleChannel, scaleType: ScaleType, orient: string, config: Config) {
+  const typeBasedConfigTypes =
```
Comment (domoritz) [inline src/compile/axis/config.ts:18]: convert to if?

Comment (domoritz) [reply src/compile/axis/config.ts:18]: The chain is really hard to read.

File: src/compile/axis/properties.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import {NOMINAL, ORDINAL, Type} from '../../type';
 import {contains, normalizeAngle} from '../../util';
 import {isSignalRef} from '../../vega.schema';
 import {UnitModel} from '../unit';
-import {getAxisConfig} from './config';
+import {AxisConfigs, getAxisConfig} from './config';
 
 // TODO: we need to refactor this method after we take care of config refactoring
```
Comment (domoritz) [inline src/compile/axis/properties.ts:16]: Still relevant

File: src/compile/axis/config.ts (hunk 1 above)
Comment (kanitw) [reply src/compile/axis/config.ts:18]: nah, I disagree for this one.

File: src/compile/axis/properties.ts (hunk 2 above)
Comment (kanitw) [reply src/compile/axis/properties.ts:16]: will remove in a subsequent refactor PR, so we don't have to rebase this one.

File: src/compile/axis/config.ts (hunk 1 above)
Comment (domoritz) [reply src/compile/axis/config.ts:18]: Why do you think this ternary is better than some ifs?

**Pull request merged**